### ISSUE TITLE
Missing Combo Box Role Implementation

### ISSUE THREAD

**Issue by ashuntu** [flutter/flutter#172918]
### Steps to reproduce

1. Create the default new Flutter application via VSCode
2. Add a `Semantics` node with `role: SemanticsRole.comboBox`
3. Run the app
5. Observe combo box role check error

### Expected results

The combo box role should be implemented so that screen readers can determine differentiate combo boxes.

It should at least just set the ATK role for Linux.

### Actual results

Combo boxes lead to errors on runtime, and do not change behavior.

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';
import 'package:flutter/semantics.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key, required this.title});

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  int _counter = 0;

  void _incrementCounter() {
    setState(() {
      _counter++;
    });
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        backgroundColor: Theme.of(context).colorScheme.inversePrimary,
        title: Text(widget.title),
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            const Text('You have pushed the button this many times:'),
            Text(
                '$_counter',
                style: Theme.of(context).textTheme.headlineMedium,
            ),
            Semantics(
              label: 'This is a semantic label',
              role: SemanticsRole.comboBox, // error for this usage
              child: TextField(
                enabled: true,
                decoration: InputDecoration(hintText: 'Input some text'),
                onChanged: (_) {},
              ),
            ),
          ],
        ),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: _incrementCounter,
        tooltip: 'Increment',
        child: const Icon(Icons.add),
      ),
    );
  }
}
```
</details>

### Logs

<details open><summary>Logs</summary>

[logs.txt](https://github.com/user-attachments/files/21496461/logs.txt)

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
$ flutter doctor -v
[✓] Flutter (Channel stable, 3.32.5, on Ubuntu 24.04.2 LTS 6.11.0-29-generic, locale en_US.UTF-8) [27ms]
    • Flutter version 3.32.5 on channel stable at /home/[USERNAME]/fvm/versions/3.32.5
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision fcf2c11572 (2 weeks ago), 2025-06-24 11:44:07 -0700
    • Engine revision dd93de6fb1
    • Dart version 3.8.1
    • DevTools version 2.45.1

[✗] Android toolchain - develop for Android devices [33ms]
    ✗ Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.dev/to/linux-android-setup for detailed instructions).
      If the Android SDK has been installed to a custom location, please use
      `flutter config --android-sdk` to update to that location.


[✗] Chrome - develop for the web (Cannot find Chrome executable at google-chrome) [12ms]
    ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a Chrome executable.

[✓] Linux toolchain - develop for Linux desktop [117ms]
    • Ubuntu clang version 18.1.3 (1ubuntu1)
    • cmake version 3.28.3
    • ninja version 1.11.1
    • pkg-config version 1.8.1
    ! Unable to access driver information using 'eglinfo'.
      It is likely available from your distribution (e.g.: apt install mesa-utils)

[!] Android Studio (not installed) [10ms]
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/to/linux-android-setup for detailed instructions).

[✓] VS Code (version unknown) [10ms]
    • VS Code at /snap/code/current/usr/share/code
    • Flutter extension can be installed from:
      🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter
    ✗ Unable to determine VS Code version.

[✓] Connected device (1 available) [37ms]
    • Linux (desktop) • linux • linux-x64 • Ubuntu 24.04.2 LTS 6.11.0-29-generic

[✓] Network resources [178ms]
    • All expected network resources are available.

! Doctor found issues in 3 categories.
```

</details>

**Comment by tirth-patel-nc** [comment]
Thanks for the report. Seeing the same behaviour with latest SDK versions on iOS. Not reproducible on Android, macOS & Web. I do not have a Linux machine to verify.

```
stable : 3.32.8
master : 3.33.0-1.0.pre-1175
```

<details><summary>Logs</summary>

```console
Restarted application in 209ms.

════════ Exception caught by scheduler library ═════════════════════════════════
The following assertion was thrown during a scheduler callback:
Missing checks for role SemanticsRole.comboBox

When the exception was thrown, this was the stack:
#0      SemanticsNode._addToUpdate.<anonymous closure> (package:flutter/src/semantics/semantics.dart:3467:9)
#1      SemanticsNode._addToUpdate (package:flutter/src/semantics/semantics.dart:3470:6)
#2      SemanticsOwner.sendSemanticsUpdate (package:flutter/src/semantics/semantics.dart:4233:14)
#3      PipelineOwner.flushSemantics (package:flutter/src/rendering/object.dart:1502:24)
#4      PipelineOwner.flushSemantics (package:flutter/src/rendering/object.dart:1504:15)
#5      RendererBinding.drawFrame (package:flutter/src/rendering/binding.dart:636:25)
#6      WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:1242:13)
#7      RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:495:5)
#8      SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1438:15)
#9      SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1351:9)
#10     SchedulerBinding.scheduleWarmUpFrame.<anonymous closure> (package:flutter/src/scheduler/binding.dart:1060:9)
#11     PlatformDispatcher.scheduleWarmUpFrame.<anonymous closure> (dart:ui/platform_dispatcher.dart:889:16)
#15     _RawReceivePort._handleMessage (dart:isolate-patch/isolate_patch.dart:193:12)
(elided 3 frames from class _Timer and dart:async-patch)
════════════════════════════════════════════════════════════════════════════════

════════ Exception caught by Flutter framework ═════════════════════════════════
'package:flutter/src/widgets/binding.dart': Failed assertion: line 1218 pos 16: 'debugFrameWasSentToEngine': is not true.
════════════════════════════════════════════════════════════════════════════════
```

</details>

**Comment by chunhtai** [comment]
This will also need to be supported for each platform.
